### PR TITLE
 :bug: Deal with missed deleted cluster events 

### PR DESCRIPTION
**Pull request by ezrasilvera**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/clustermanager/reconcile_logicalcluster.go [hunk 1]
```diff
@@ -64,22 +64,22 @@ func (c *controller) processAddOrUpdateLC(logicalCluster *lcv1alpha1.LogicalClus
 			LogicalClusters(GetNamespace(logicalCluster.Spec.ClusterProviderDescName)).
 			Delete(c.context, logicalCluster.Name, v1.DeleteOptions{})
 	}
+	providerInfo, err := c.clientset.LogicalclusterV1alpha1().ClusterProviderDescs().Get(
```
Comment (andreyod) [inline pkg/clustermanager/reconcile_logicalcluster.go:67]: Get the provider from cache here.

File: pkg/clustermanager/reconcile_logicalcluster.go [hunk 2]
```diff
@@ -97,18 +97,22 @@ func (c *controller) processAddOrUpdateLC(logicalCluster *lcv1alpha1.LogicalClus
...
-	// case lcv1alpha1.LogicalClusterPhaseReady:
-	// The cluster is ready - we have nothing to do but celebrate!
-	// Likely we got here after the provider finished creating a managed
-	// cluster and dicovery moved it into the Ready state.
-	//
-	// If a logical cluster has been created for an unmanaged physical
-	// cluster, then wait for discovery to move the phase to Ready.
+
+	// The code from here deals with the case of provider deleted cluster events that might be missed
+	_, err = provider.Get(c.context, logicalCluster.Name)
```
Comment (andreyod) [inline pkg/clustermanager/reconcile_logicalcluster.go:102]: This is done on each event? What is the performance hit?

Comment (effi-ofer) [reply pkg/clustermanager/reconcile_logicalcluster.go:102]: General question - how frequent are LogicalCluster events? I would have thought that they occur only on cluster changes (which is infrequent) and on startup.  And what's the expected number of clusters?

**Comment by effi-ofer** [comment]
Is it possible to lose a delete event if we have a finalizer?

File: pkg/clustermanager/reconcile_logicalcluster.go (hunk 2 above)
Comment (ezrasilvera) [reply pkg/clustermanager/reconcile_logicalcluster.go:102]: This will be taken from a cache I'm building

**Comment by clubanderson** [comment]
please review and close as snowflake if no longer valid

**Comment by effi-ofer** [comment]
> please review and close as snowflake if no longer valid

I think this is still needed but the branch needs to be changed to main and the code rebased.

**Comment by ezrasilvera** [comment]
not  relevant for ks-0.20

**Pull request closed without merging**